sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -540,7 +540,11 @@ async def _read(self, toread, buffer=None, **kwargs):  # pylint: disable=unused-
                     self._read_buffer = BytesIO(data[toread:])
                     toread = 0
             return view
-        except:
+        except TypeError as te:
+            # aiohttp websocket raises TypeError when a websocket disconnects, as it ends up
+            # reading None over the wire and cant convert to bytes.
+            raise ConnectionError("Websocket disconnected: %r" % te) from None
```

Comment:
Where does this ConnectionError get caught -- retry loop in receive_task?

Reply:
`ConnectionError` is a subclass of `OSError`, so it heads up the chain within transport (read or write ) and ultimately all the way to the retry loop.